src/components/common/PaginatedTemplateList/PaginatedTemplateList.tsx

```diff
@@ -0,0 +1,72 @@
+"use client";
+
+import { useState } from "react";
+
+import Pagination from "../Pagination/Pagination";
+import TemplateListItem from "../TemplateListItem/TemplateListItem";
+
+interface PaginatedTemplateListProperties {
+  templates: {
+    id: number;
+    name: string;
+    image: string;
+  }[];
+  onPreview: (id: number) => void;
+  itemsPerPage: number;
+}
+
+const PaginatedTemplateList = ({
+  templates,
```

Comment:
what happens if templates is undefined or is not an array?

Reply:
condition handled